### PR TITLE
Add packaging functionality

### PR DESCRIPTION
**Pull request by gaborod16**
Integration of packaging capabilities for portability

### CODE REVIEW

File: docker-compose.published_dependencies.yml [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    command:
+      - '--config.file=/etc/prometheus/prometheus.yml'
+      - '--storage.tsdb.path=/prometheus'
+      - '--web.console.libraries=/etc/prometheus/console_libraries'
+      - '--web.console.templates=/etc/prometheus/consoles'
+      - '--storage.tsdb.retention.time=200h'
+      - '--web.enable-lifecycle'
+    restart: unless-stopped
+    expose:
+      - 9090
```
Comment (teddius) [inline docker-compose.published_dependencies.yml]: Add port into environment file

File: docker-compose.published_dependencies.yml [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    image: ${PROD_ALERT_MANAGER_IMAGE_NAME}
+    container_name: alertmanager
+    volumes:
+      - ./alertmanager:/etc/alertmanager
+    command:
+      - '--config.file=/etc/alertmanager/config.yml'
+      - '--storage.path=/alertmanager'
+    restart: unless-stopped
+    expose:
+      - 9093
```
Comment (teddius) [inline docker-compose.published_dependencies.yml]: Add port into environment file

File: docker-compose.published_dependencies.yml [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+      - /sys:/host/sys:ro
+      - /:/rootfs:ro
+    command:
+      - '--path.procfs=/host/proc'
+      - '--path.rootfs=/rootfs'
+      - '--path.sysfs=/host/sys'
+      - '--collector.filesystem.mount-points-exclude=^/(sys|proc|dev|host|etc)($$|/)'
+    restart: unless-stopped
+    expose:
+      - 9100
```
Comment (teddius) [inline docker-compose.published_dependencies.yml]: Add port into environment file

File: docker-compose.published_dependencies.yml [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+    container_name: cadvisor
+    volumes:
+      - /:/rootfs:ro
+      - /var/run:/var/run:rw
+      - /sys:/sys:ro
+      - /var/lib/docker:/var/lib/docker:ro
+      #- /cgroup:/cgroup:ro #doesn't work on MacOS only for Linux
+    restart: unless-stopped
+    expose:
+      - 8080
```
Comment (teddius) [inline docker-compose.published_dependencies.yml]: Add port into environment file

File: docker-compose.published_dependencies.yml [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+      - grafana_data:/var/lib/grafana
+      - ./grafana/provisioning/dashboards:/etc/grafana/provisioning/dashboards
+      - ./grafana/provisioning/datasources:/etc/grafana/provisioning/datasources
+    environment:
+      - GF_SECURITY_ADMIN_USER=${ADMIN_USER:-admin}
+      - GF_SECURITY_ADMIN_PASSWORD=${ADMIN_PASSWORD:-admin}
+      - GF_USERS_ALLOW_SIGN_UP=false
+    restart: unless-stopped
+    expose:
+      - 3000
```
Comment (teddius) [inline docker-compose.published_dependencies.yml]: Add port into environment file

File: docker-compose.published_dependencies.yml [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+      - monitor-net
+    labels:
+      org.label-schema.group: "monitoring"
+
+  pushgateway:
+    image: ${PROD_PUSH_GATEWAY_IMAGE_NAME}
+    container_name: pushgateway
+    restart: unless-stopped
+    expose:
+      - 9091
```
Comment (teddius) [inline docker-compose.published_dependencies.yml]: Add port into environment file

File: docker-compose.published_dependencies.yml [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+    networks:
+      - monitor-net
+    labels:
+      org.label-schema.group: "monitoring"
+
+  caddy:
+    image: ${PROD_CADDY_IMAGE_NAME}
+    container_name: caddy
+    ports:
```
Comment (teddius) [inline docker-compose.published_dependencies.yml:113]: Add ports into environment file

File: packaging/example.env [hunk 8]
```diff
@@ -0,0 +1,38 @@
+# Packaging
```
Comment (teddius) [inline packaging/example.env:4]: Add ports into this environment file so it is possible to configure the ports from the environment file

**Review by teddius — CHANGES_REQUESTED** [review]

**Review by MohamedYaman — APPROVED** [review]
Good job Gabe!

**Pull request merged**